pkg/cloud/aws/services/userdata/controlplane.go

```diff
@@ -157,14 +213,65 @@ type ContolPlaneJoinInput struct {
 	CACertHash     string
 	CACert         string
 	CAKey          string
+	EtcdCert       string
+	EtcdKey        string
+	FrontProxyCert string
+	FrontProxyKey  string
+	SaCert         string
+	SaKey          string
 	BootstrapToken string
 	ELBAddress     string
-	KubeConfig     string
+}
+
+func (cpi *ControlPlaneInput) isValid() (bool, string) {
+	// TODO: ashish-amarnath verify if ca cert and key is mandatory for kubeadm init
+	if cpi.CACert == "" || cpi.CAKey == "" {
+		return false, "CA cert material in the ControlPlaneInput is invalid"
+	}
+
+	if !isKeyPairValid(cpi.EtcdCert, cpi.EtcdKey) {
+		return false, "ETCD cert material in the ControlPlaneInput is invalid"
+	}
+
+	if !isKeyPairValid(cpi.FrontProxyCert, cpi.FrontProxyKey) {
+		return false, "FrontProxy cert material in ControlPlaneInput is invalid"
+	}
+
+	if !isKeyPairValid(cpi.SaCert, cpi.SaKey) {
+		return false, "ServiceAccount cert material in ControlPlaneInput is invalid"
+	}
+
+	return true, ""
+}
+
+func (cpi *ContolPlaneJoinInput) isValid() (bool, string) {
+	if !isKeyPairValid(cpi.CACert, cpi.CAKey) {
+		return false, "CA cert material in the ContolPlaneJoinInput is invalid"
+	}
+
+	if !isKeyPairValid(cpi.EtcdCert, cpi.EtcdKey) {
+		return false, "ETCD cert material in the ContolPlaneJoinInput is invalid"
+	}
+
+	if !isKeyPairValid(cpi.FrontProxyCert, cpi.FrontProxyKey) {
+		return false, "FrontProxy cert material in ContolPlaneJoinInput is invalid"
+	}
+
+	if !isKeyPairValid(cpi.SaCert, cpi.SaKey) {
+		return false, "ServiceAccount cert material in ContolPlaneJoinInput is invalid"
+	}
+
+	return true, ""
 }
 
 // NewControlPlane returns the user data string to be used on a controlplane instance.
 func NewControlPlane(input *ControlPlaneInput) (string, error) {
 	input.Header = defaultHeader
+	valid, reason := input.isValid()
```

Comment:
This will return an error instead and then we can simply bubble the error up, there's no reason to wrap any further here unless additional context is required

Reply:
fixed.